builtin/logical/pki/path_roles.go

```diff
@@ -339,7 +339,7 @@ func (b *backend) pathRoleCreate(
 		// If they are using the system default, cap it to the role max;
 		// if it was specified on the command line, make it an error
 		if len(entry.TTL) == 0 {
-			ttl = maxTTL
+			entry.TTL = entry.MaxTTL
```

Comment:
On line 318, we're setting up `maxTTL`.  On line 342 we've evaluated the TTL and need to write the resulting TTL back to `entry` in order to have the value written out on line 355.  Right now no TTL is being saved along with the logical backend when TTL exceeds `maxTTL`.  Am I missing something?